service/src/java/org/apache/hive/service/server/HiveServer2.java

```diff
@@ -443,11 +463,19 @@ public synchronized void init(HiveConf hiveConf) {
           }
           builder.addServlet("llap", LlapServlet.class);
           builder.addServlet("jdbcjar", JdbcJarDownloadServlet.class);
-          builder.setContextRootRewriteTarget("/hiveserver2.jsp");
+          builder.setContextRootRewriteTarget(HS2_WEBUI_ROOT_URI);
 
           webServer = builder.build();
           webServer.addServlet("query_page", "/query_page.html", QueryProfileServlet.class);
           webServer.addServlet("api", "/api/*", QueriesRESTfulAPIServlet.class);
+          if (hiveConf.getBoolVar(ConfVars.HIVE_SERVER2_WEBUI_ENABLE_LDAP)) {
+            if (passwdAuthenticationProvider == null)
```

Comment:
missing curly braces in inner if/else

Reply:
Done

Reply:
almost there, in java it's:
```
} else {
```

instead of:
```
 }
else {
```

Reply:
Done